drivers/can/can_mcp2515.c

```diff
@@ -352,6 +364,18 @@ static int mcp2515_configure(struct device *dev, enum can_mode mode,
 		goto done;
 	}
 
+	/* Wait 128 OSC1 clock cycles at 1MHz (minimum clock in frequency)
+	* see MCP2515 datasheet section 8.1 Oscillator Start-up Timer
+	*/
+	k_usleep(128);
+
+	ret = mcp2515_get_mode(dev);
+	if (ret != MCP2515_MODE_CONFIGURATION) {
+		LOG_ERR("Device did not reset into configuration mode [%d]",
+			ret);
+		return -EIO;
```

Comment:
For cleanliness `goto done` here to clean up the locked mutex. 
Also I just noticed that `done` label sits in the wrong spot, should be further down before the mutex release.  `goto done` is used inconsistently in the rest of this function as well. I will make a small patch to fix the handling.